### PR TITLE
Dependencies: Fixed numpy v2 support on Windows; Continous integration: Added tests with numpy v2; Fixed test by using PySide<6.7

### PR DESCRIPTION
**Pull request by t20100**
Checklist:

- [x] The PR title is formatted as: `<Module or Topic>: <Action> <Summary>` (see [contributing guidelines](https://github.com/silx-kit/silx/blob/main/CONTRIBUTING.rst#pull-request-title-format))

<hr>

This PR aims at having CI pass

### CODE REVIEW

**Comment by t20100** [comment]
This PR also fixes the cython code so that tests pass on Windows with numpy 2, since there is a breaking change regarding [Windows default integer](https://numpy.org/devdocs/numpy_2_0_migration_guide.html#windows-default-integer)

**Comment by t20100** [comment]
This PR also modifies: 
- The `strip` fit filter to use `int` instead of `long` (I expect int64 is not needed for indices here...): This avoids the difference between Windows and others regarding the definition of `long` 
- The median filter to use sized types to avoid. Again an issue with `long` on Windows.

**Comment by t20100** [comment]
This is ready for review.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -120,15 +120,7 @@ jobs:
...
-          if [ "${{ matrix.QT_BINDING }}" == "PyQt5" ]; then
-            pip install --pre pyqt5;
-          fi
-          if [ "${{ matrix.QT_BINDING }}" == "PySide6" ]; then
-            pip install --pre pyside6;
-          fi
-          if [ "${{ matrix.QT_BINDING }}" == "PyQt6" ]; then
-            pip install --pre pyqt6;
-          fi
+          pip install --pre "${{ matrix.QT_BINDING }}"
```
Comment (payno) [inline .github/workflows/ci.yml:123]: This is the kind of commit I like ;)

**Pull request merged**